firedrake/preconditioners/pmg.py

```diff
@@ -262,6 +262,27 @@ def prolongation_transfer_kernel_aij(Pk, P1):
     return kernel
 
 
+def isTensorProductSpace(V):
+    ele = V.ufl_element()
+    cell = ele.cell()
+    isquad = cell.cellname().find("quadrilateral") >= 0
+
+    if isinstance(ele, (firedrake.VectorElement, firedrake.TensorElement)):
+        subel = ele.sub_elements()
+        ele = subel[0]
+
+    N = ele.degree()
+    if isinstance(ele, firedrake.TensorProductElement):
+        subel = ele.sub_elements()
+        ele = subel[0]
+        N = N[0]
```

Comment:
Does this logic do the right thing if I take e.g. the tensor product of CG(k) on quads and Hermite(3) on the interval?